ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualTransactionBuilder.kt

```diff
@@ -38,24 +38,8 @@ interface ConsensualTransactionBuilder {
      *
      * @throws IllegalStateException when called a second time on the same object to prevent
      *      unintentional duplicate transactions.
+     * @throws CordaRuntimeException if none of the required keys are available to sign the transaction.
```

Comment:
Feels like this should be of a more specific exception type?

Reply:
I am a bit uncertain about this one.
We should not have SignedTxs without signatures.
But the TransactionSIgnature service should not fail if there are no keys. That can just return an empty list.
So I ended up testing if any signatures were generated in the SignedTxFactories and throwing this exception if there were none.